src/transaction_builder.js

```diff
@@ -227,14 +465,39 @@ export class TransactionBuilder {
       this.timebounds.maxTime = this.timebounds.maxTime.getTime() / 1000;
     }
 
+    attrs.preconditions = {};
+
     this.timebounds.minTime = UnsignedHyper.fromString(
       this.timebounds.minTime.toString()
     );
     this.timebounds.maxTime = UnsignedHyper.fromString(
       this.timebounds.maxTime.toString()
     );
+    attrs.preconditions.timeBounds = new xdr.TimeBounds(this.timebounds);
```

Comment:
This will be clearer once you merge in the XDR changes, but the behavior we need is: if **only** timebounds are set, then set `attrs.cond = xdr.Preconditions.precondTime(this.timebounds)` (that code will already exist after rebase), otherwise, use `precondV2`

Reply:
Why not always use a precondV2?

Reply:
Couple reasons (@leighmcculloch and I discussed this at length during the design phase): 
 - bwd compat: we want transactions built before this SDK release and after this SDK release to result in the same XDR
 - optimization: V2 w/ only timebounds forms a larger binary because of the other (empty) fields